src/qutip_qip/device/spinchain.py

```diff
@@ -52,44 +52,53 @@ class SpinChain(ModelProcessor):
     The available Hamiltonian of the system is predefined.
     The processor can simulate the evolution under the given
     control pulses either numerically or analytically.
-    It is a base class and should not be used directly, please
-    refer the the subclasses :class:`.LinearSpinChain` and
-    :class:`.CircularSpinChain`.
-    (Only additional attributes are documented here, for others please
-    refer to the parent class :class:`.ModelProcessor`)
+    This is the base class and should not be used directly.
+    Please use :class:`.LinearSpinChain` or :class:`.CircularSpinChain`.
 
     Parameters
     ----------
     num_qubits: int
         The number of qubits in the system.
 
-    correct_global_phase: float
+    correct_global_phase: float, optional
         Save the global phase, the analytical solution
         will track the global phase.
         It has no effect on the numerical solution.
 
-    sx: int or list
-        The delta for each of the qubits in the system.
-
-    sz: int or list
-        The epsilon for each of the qubits in the system.
-
-    sxsy: int or list
-        The interaction strength for each of the qubit pair in the system.
-
-    t1: list or float
+    t1: list or float, optional
         Characterize the decoherence of amplitude damping for
-        each qubit. A list of size `num_qubits` or a float for all qubits.
+        each qubit. A list of size ``num_qubits`` or a float for all qubits.
 
-    t2: list of float
+    t2: list of float, optional
         Characterize the decoherence of dephasing for
-        each qubit. A list of size `num_qubits` or a float for all qubits.
+        each qubit. A list of size ``num_qubits`` or a float for all qubits.
+
+    **params:
+        Keyword argument for hardware parameters, in the unit of frequency
+        (MHz, GHz etc, the unit of time list needs to be adjusted accordingly).
+        Qubit parameters can either be a float or an array of the length
+        ``num_qubits``.
+        ``sxsy``, should be either a float or an array of the length
+        ``num_qubits-1`` (for :class:`.LinearSpinChain`) or ``num_qubits``
```

Comment:
```suggestion
        ``num_qubits-1`` (for :class:`.LinearSpinChain`) or ``num_qubits``
```
```suggestion
        ``num_qubits=1`` (for :class:`.LinearSpinChain`) or ``num_qubits``
```

Reply:
No, it is `num_qubits-1`, for LinearSpinChain the list of coupling strength is one shorter than the number of qubits